uat/testing-features/src/main/java/com/aws/greengrass/platforms/PlatformResolver.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.platforms;
+
+import lombok.extern.log4j.Log4j2;
+import org.apache.commons.io.IOUtils;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Used by {@link Platform} to decide what platform is being tested.
+ */
+@Log4j2
+public final class PlatformResolver {
+    private static final Set<String> SUPPORTED_PLATFORMS = Collections.unmodifiableSet(Stream.of(
+            "all", "any", "unix", "posix", "linux", "debian", "windows", "fedora", "ubuntu", "macos",
+            "raspbian", "qnx", "cygwin", "freebsd", "solaris", "sunos").collect(Collectors.toSet()));
+
+    // This needs to come before RANKS so that IS_WINDOWS is declared and set
+    public static final boolean IS_WINDOWS = System.getProperty("os.name").toLowerCase().contains("wind");
+
+    public static final AtomicReference<Map<String, Integer>> RANKS =
+            new AtomicReference<>(Collections.unmodifiableMap(initializeRanks()));
+
+    private PlatformResolver() {
+    }
+
+    @SuppressWarnings("PMD.CognitiveComplexity")
+    private static Map<String, Integer> initializeRanks() {
+        Map<String, Integer> ranks = new HashMap<>();
+        // figure out what OS we're running and add applicable tags
+        // The more specific a tag is, the higher its rank should be
+        // TODO: use better way to determine if a field is platform specific. Eg: using 'platform$' prefix.
+        ranks.put("all", 0);
+        ranks.put("any", 0);
+        if (Files.exists(Paths.get("/bin/bash")) || Files.exists(Paths.get("/usr/bin/bash"))) {
+            ranks.put("unix", 3);
+            ranks.put("posix", 3);
+        }
+        if (Files.exists(Paths.get("/proc"))) {
+            ranks.put("linux", 10);
+        }
+        if (Files.exists(Paths.get("/usr/bin/apt-get"))) {
+            ranks.put("debian", 11);
+        }
+        if (IS_WINDOWS) {
+            ranks.put("windows", 5);
+        }
+        if (Files.exists(Paths.get("/usr/bin/yum"))) {
+            ranks.put("fedora", 11);
+        }
+        if (!IS_WINDOWS) {
```

Comment:
I guess we need to remove unnecessary checks
```
if (IS_WINDOWS) {
  ranks.put("windows", 5);
} else {
}
```

Reply:
extra checks are removed in f820a556c989ecf5ecb2a117bc62f3124ae7d774